analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -128,14 +129,122 @@ public static SyntaxNode WalkUpParentheses(this SyntaxNode node)
+                    _ => null,
+                };
+                if (TupleExpressionSyntaxWrapper.IsInstance(otherSide) || DeclarationExpressionSyntaxWrapper.IsInstance(otherSide))
+                {
+                    var indexAndCount = IndexAndCountOfNesting(node);
+                    return FindMatchingNestedNode(indexAndCount, otherSide);
+                }
+            }
+
+            return null;
+
+            static Stack<IndexCountPair> IndexAndCountOfNesting(SyntaxNode node)
+            {
+                Stack<IndexCountPair> indexAndCount = new();
+                while (TupleExpressionSyntaxWrapper.IsInstance(node?.Parent) || ParenthesizedVariableDesignationSyntaxWrapper.IsInstance(node?.Parent))
+                {
+                    node = node switch
+                    {
+                        ArgumentSyntax tupleArgument when TupleExpressionSyntaxWrapper.IsInstance(node.Parent) =>
+                            PushIndexAndCountTuple(indexAndCount, (TupleExpressionSyntaxWrapper)node.Parent, tupleArgument),
+                        _ when VariableDesignationSyntaxWrapper.IsInstance(node) && ParenthesizedVariableDesignationSyntaxWrapper.IsInstance(node.Parent) =>
+                            PushIndexAndCountParenthesizedDesignation(indexAndCount, (ParenthesizedVariableDesignationSyntaxWrapper)node.Parent, (VariableDesignationSyntaxWrapper)node),
+                        _ => null,
+                    };
+                    if (DeclarationExpressionSyntaxWrapper.IsInstance(node?.Parent) && node is { Parent.Parent: ArgumentSyntax { } argument })
+                    {
+                        node = argument;
+                    }
+                }
+                return indexAndCount;
+            }
+
+            static SyntaxNode FindMatchingNestedNode(Stack<IndexCountPair> indexAndCount, SyntaxNode node)
+            {
+                while (node is not null && indexAndCount.Count > 0)
+                {
+                    if (DeclarationExpressionSyntaxWrapper.IsInstance(node))
+                    {
+                        node = ((DeclarationExpressionSyntaxWrapper)node).Designation;
+                    }
+                    var indexCountPair = indexAndCount.Pop();
+                    node = node switch
+                    {
+                        _ when TupleExpressionSyntaxWrapper.IsInstance(node) => StepDownInTuple((TupleExpressionSyntaxWrapper)node, indexCountPair),
+                        _ when ParenthesizedVariableDesignationSyntaxWrapper.IsInstance(node) =>
+                            StepDownInParenthesizedVariableDesignation((ParenthesizedVariableDesignationSyntaxWrapper)node, indexCountPair),
+                        _ => null,
+                    };
+                }
+                return node;
+            }
+
+            static SyntaxNode PushIndexAndCountTuple(Stack<IndexCountPair> indexAndCount, TupleExpressionSyntaxWrapper tuple, ArgumentSyntax argument)
```

Comment:
```suggestion
            static SyntaxNode PushPathPosition(Stack<IndexCountPair> indexAndCount, TupleExpressionSyntaxWrapper tuple, ArgumentSyntax argument)
```

Reply:
I kept the suffices for consistency with the other methods.